### PR TITLE
Shipping section

### PR DESCRIPTION
**Pull request by AvnerGuyAvrahamCom**

### CODE REVIEW

File: src/components/common/ModalComponent.tsx [hunk 1]
```diff
@@ -35,27 +42,35 @@ const ModalComponent: FC<IModalProps> = ({
...
-                borderRadius={{ md: "sm", base: undefined }}
-                h={{ md: "80vh", base: "100vh" }}>
+                borderRadius={{ md: "md", base: undefined }}
+                h={{ md: "90vh", base: "100vh" }}
+              >
                 <Flex
                   h="full"
                   direction={{ md: "row-reverse", base: "column" }}
-                  justifyContent="space-between">
+                  justifyContent="space-between"
```
Comment (artemGuyAvrahamCom) [inline src/components/common/ModalComponent.tsx:72]: Change justifyContent to justify

File: src/components/CheckoutModal/OrderSummary/GeneralDetails.tsx [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
-import { Box, Flex } from "@chakra-ui/react";
+import { Flex, Text } from "@chakra-ui/react";
 
 interface IGeneralDetail {
   name: string;
   value: number | undefined;
   isBold?: boolean;
   currency?: string;
+  isGray?: boolean;
 }
```
Comment (artemGuyAvrahamCom) [inline src/components/CheckoutModal/OrderSummary/GeneralDetails.tsx:10]: Move all Interfaces into 1 file. For example types.ts

File: src/components/common/ModalComponent.tsx (hunk 1 above)
Comment (artemGuyAvrahamCom) [inline src/components/common/ModalComponent.tsx:72]: Change justifyContent to justify

File: src/components/CheckoutModal/OrderSummary/InputSummary.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+} from "@chakra-ui/react";
+
+interface IInputSummary {
+  offerName: string;
+  inputValue: string;
+  handleChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  setIsInputVisible: Dispatch<SetStateAction<boolean>>;
+  isInputVisible: boolean;
+  isInvalid: boolean;
+}
```
Comment (artemGuyAvrahamCom) [inline src/components/CheckoutModal/OrderSummary/InputSummary.tsx:23]: Move all interfaces into another file. For example types.ts. It makes code cleaner

**Review by artemGuyAvrahamCom — CHANGES_REQUESTED** [review]

File: src/components/CheckoutModal/OrderSummary/InputSummary.tsx (hunk 3 above)
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/InputSummary.tsx:23]: done

File: src/components/CheckoutModal/OrderSummary/GeneralDetails.tsx (hunk 2 above)
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/GeneralDetails.tsx:10]: done

File: src/components/common/ModalComponent.tsx [hunk 4]
```diff
@@ -35,27 +42,35 @@ const ModalComponent: FC<IModalProps> = ({
             justify="center"
             position="fixed"
             top={0}
-            bgColor="blackAlpha.800">
-            <Box w={{ md: "70vw", base: "full" }}>
+            bgColor="blackAlpha.800"
+          >
+            <Box w={{ md: "880px", base: "full" }}>
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/common/ModalComponent.tsx:47]: done

File: src/components/CheckoutModal/OrderSummary/SpecialOffer.tsx [hunk 5]
```diff
@@ -32,52 +22,74 @@ export const SpecialOffer: FC<ISpecialOffer> = ({ offerName }) => {
...
+            isInputVisible={isInputVisible}
+            isInvalid={isInvalid}
+          />
+          {inputValue && !isInvalid && (
+            <Button
+              ml="1"
+              w="fit-content"
+              variant="unstyled"
+              fontSize="11"
+              padding="0px"
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/SpecialOffer.tsx:69]: done

File: src/components/CheckoutModal/OrderSummary/SpecialOffer.tsx [hunk 6]
```diff
@@ -32,52 +22,74 @@ export const SpecialOffer: FC<ISpecialOffer> = ({ offerName }) => {
...
+              onClick={handleApply}
+            >
               APPLY
             </Button>
+          )}
+          {isInvalid && (
             <Button
+              right="0"
+              position="absolute"
+              justifyContent="right"
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/SpecialOffer.tsx:82]: done

File: src/components/common/ModalComponent.tsx (hunk 1 above)
Comment (AvnerGuyAvrahamCom) [reply src/components/common/ModalComponent.tsx:72]: done

File: src/components/CheckoutModal/OrderSummary/SpecialOffer.tsx [hunk 7]
```diff
@@ -32,52 +22,74 @@ export const SpecialOffer: FC<ISpecialOffer> = ({ offerName }) => {
...
+    <Box my="4">
       {!isInputVisible && (
         <Button
+          my="3"
           onClick={handleClick}
-          variant="unstyled"
+          variant="summary"
           w="full"
           leftIcon={<PlusSquareIcon />}
+          justifyContent="flexStart"
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/SpecialOffer.tsx:48]: done

File: src/components/CheckoutModal/OrderSummary/GeneralDetails.tsx [hunk 8]
```diff
@@ -1,28 +1,33 @@
...
 
 export const GeneralDetail: FC<IGeneralDetail> = ({
   name,
   value,
   isBold,
   currency,
+  isGray,
 }) => (
   <Flex
     justifyContent="space-between"
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/GeneralDetails.tsx:20]: done

File: src/components/CheckoutModal/OrderSummary/InputSummary.tsx [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+          <Input
+            ref={inputRef}
+            variant="unstyled"
+            autoFocus
+            value={inputValue}
+            onChange={handleChange}
+            onFocus={() => setIsOnFocus(true)}
+            pl="0"
+            pr="12"
+            borderRadius="0px"
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/InputSummary.tsx:63]: done

File: src/components/CheckoutModal/OrderSummary/InputSummary.tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      (!inputRef.current?.value && setIsInputVisible(false)) ||
+      setIsOnFocus(false),
+  });
+
+  const isFocusLabel = isOnFocus ? "blackAlpha.900" : "gray.500";
+  const borderBottom = isInvalid
+    ? "2px solid red"
+    : isOnFocus
+    ? "2px"
+    : "2px solid gray";
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/InputSummary.tsx:48]: done

File: src/components/CheckoutModal/OrderSummary/index.tsx [hunk 11]
```diff
@@ -33,16 +39,19 @@ const OrderSummary = () => {
...
           name="Delivery"
           value={orderData.delivery}
           currency={orderData.currency}
+          isGray
         />
         <GeneralDetail
           name="Taxes"
           value={orderData.taxes}
           currency={orderData.currency}
+          isGray
```
[the remark this answers is not in the document]
Comment (AvnerGuyAvrahamCom) [reply src/components/CheckoutModal/OrderSummary/index.tsx]: done

**Review by artemGuyAvrahamCom — APPROVED** [review]
good

**Pull request merged**